cli/lsp/language_server.rs

```diff
@@ -2982,6 +2982,27 @@ impl Inner {
 
 #[tower_lsp::async_trait]
 impl tower_lsp::LanguageServer for LanguageServer {
+  async fn execute_command(
+    &self,
+    params: ExecuteCommandParams,
+  ) -> LspResult<Option<Value>> {
+    if params.command == "deno.cache" {
+      // We're sending Vec<Vec<uris_to_cache>> in response to deno cache code action
+      let uris: Vec<Vec<String>> = params
+        .arguments
+        .into_iter()
+        .map(|v| serde_json::from_value::<Vec<String>>(v))
+        .map(|v| v.map_err(|e| tower_lsp::jsonrpc::Error::invalid_request())) // FIXME: handle the error
+        .collect::<LspResult<Vec<_>>>()?;
+      return self.cache_request(Some(json!({
+        // TODO: this is wrong, how to get the referrer
+        "referrer": TextDocumentIdentifier{ uri: Url::parse(&uris[0][0]).unwrap()},
```

Comment:
@bartlomieju  The cache request needs the referrer here, as far as I understand it the referrer is the current open document

I'm currently just using the uris which contains the url that the user is going to its definition which is incorrect

The way it works now as far as I can tell:
- uesr hits cache code action
- lsp server responds with the uris to cache
- lsp client responds with the referrer (which is the current active document) and the uris to cache (got it from the server)

maybe the lsp server in step 2 can send the referrer as well (If the info is available) ?

Reply:
@nayeemrmn maybe you could advise here? :)